packages/carbon-component-mapper/src/text-field/text-field.js

```diff
@@ -23,6 +27,7 @@ const TextField = (props) => {
       invalidText={invalid || ''}
       warn={Boolean(warn)}
       warnText={warn || ''}
+      onChange={(e) => setValue(e, input)}
```

Comment:
you have to call `input.onChange(e)` somewhere to change the value in the form state. see https://data-driven-forms.org/hooks/use-field-api#onchange